app/models/health_care_application.rb

```diff
@@ -26,13 +26,8 @@ class HealthCareApplication < ApplicationRecord
   validates(:form, presence: true, on: :create)
   validate(:form_matches_schema, on: :create)
 
-  after_save(:send_failure_mail, if: proc do |hca|
-    hca.saved_change_to_attribute?(:state) && hca.failed? && hca.form.present? && hca.parsed_form['email']
```

Comment:
Discovered we weren't testing this logic, so I added specs for them. As part of that I pulled out the conditionals to some methods to better DRY it up and make it more readable.